cdk/lib/security-hq.ts

```diff
@@ -118,6 +121,23 @@ dpkg -i /tmp/installer.deb`,
       },
     });
 
+    // TODO reduce TTL and point to new ALB after going live.
+    const oldElb = template.getResource('LoadBalancer') as CfnLoadBalancer;
+
+    this.withStageDependentValue({
+      variableName: 'domain',
+      stageValues: {
+        [Stage.CODE]: 'security-hq.code.dev-gutools.co.uk',
+        [Stage.PROD]: 'security-hq.gutools.co.uk',
+      },
+    });
+
+    new GuCname(this, 'security-hq.gutools.co.uk', {
+      name: this.mappings.findInMap(this.stage, 'domainName'),
```

Comment:
`withStageDependentValue` returns a `T` (`string` in this case), so we can simplify this a little ([inspiration](https://github.com/guardian/cdk/blob/624f45762ec875a905c7d8759166ce0f20e747fd/src/constructs/dns/dns-records.ts#L72-L79)):

```typescript
const domainName = this.withStageDependentValue({
  variableName: 'domain'Name,
  stageValues: {
    [Stage.CODE]: 'security-hq.code.dev-gutools.co.uk',
    [Stage.PROD]: 'security-hq.gutools.co.uk',
  },
});

new GuCname(this, 'security-hq.gutools.co.uk', {
  name: domainName,
  ttl: Duration.hours(1),
  resourceRecord: oldElb.attrDnsName,
});
```

This might suggest that `withStageDependentValue`'s docs need refining to improve clarity.

Looking at the snapshot diff, I think we can re-use the `domainName` mapping value too, rather than create a new `domain` mapping.

Reply:
Alternatively if you upgrade the `@guardian/cdk` version you can use the updated version of this construct: https://github.com/guardian/cdk/pull/912#issuecomment-968930718.

Reply:
@akash1810 ah nice, I did wonder if something like this was possible!

Re: re-using domainName (from ec2/certificate), I considered this but was a bit wary, as thought it could be confusing/a bit magical.

Reply:
(Though realise, I'd inadvertently done this by `findInMap(this.stage, 'domainName'),`, which was an accident.)

Reply:
@jacobwinch nice! That is a lot cleaner.